### ISSUE TITLE
Let users specify component types in TestEventReceiver

### ISSUE THREAD

**Issue by kaen** [Terasology/ModuleTestingEnvironment#4]
Currently, the `TestEventReceiver` allows to specify the event class to be caught, but it lacks the abilitiy to restrict event handling to a specific set of components to be present. This, however, is crucial for events such as `OnAddedComponent` which themselves don't carry any information. Without the indication of component classes it is impossible to react to these kind of "informative events" that rely on the component check.

> Note that this event will only be received by @ReceiveEvent methods where all components in its list are present and at least one is involved in the action causing the event.

As the components are just specified as list of component class it may be possible to extend the existing interface with another parameter taking this list:

```java
TestEventReceiver receiver = new TestEventReceiver<>(context, OnComponentAdded.class, (event, entity) -> {
  // do something with the event or entity
}, List.of(StunEffect.class));
```

This receiver signature should be equivalent to the following event handler:

```java
@ReceiveEvent(components = {StunEffect.class})
public void onStunAdded(OnComponentAdded event, EntityRef entity) {
  // do something with the event or entity
}
```

~Blocked by https://github.com/MovingBlocks/Terasology/pull/3147~

**Comment by Cervator** [comment]
No longer blocked! Code away :-)

**Comment by skaldarnar** [comment]
Updated the issue description with more information and motivation.

**Comment by kaen** [comment]
Fixed in #19 :tada:
